bundler/lib/bundler/remote_specification.rb

```diff
@@ -97,12 +97,48 @@ def git_version
       " #{source.revision[0..6]}"
     end
 
+    # we don't get the checksum from a server like we could with EndpointSpecs
+    # calculating the checksum from the file on disk still provides some measure of security
+    # if it changes from install to install, that is cause for concern
+    def checksum
+      @checksum ||= begin
+        return unless _fetched_gem
+
+        File.open(_fetched_gem) do |f|
+          digest = Bundler::SharedHelpers.digest(:SHA256).new
```

Comment:
can we share this with the code in `RubyGemsGemInstaller#validate_bundler_checksum` ?